### PR TITLE
Marina X psetv2

### LINKED ISSUE #401: liquid.taxi: Add support for new version with psetv2

### PR DESCRIPTION
**Pull request by louisinger**
**This PR is using new psetv2 API and new taxi (supporting psetv2)**

this is a WIP branch focused on taxi case mainly for testing the new pset. Further works are needed in order to support pset v2 everywhere in marina. cc @altafan 

it closes #401

### CODE REVIEW

File: src/application/utils/transaction.ts [hunk 1]
```diff
@@ -234,14 +266,26 @@ export function createTaxiTxFromTopup(
...
+  const updater = new Updater(pset);
+  updater.addInputs(selectedUtxos.map((u) => new CreatorInput(u.txid, u.vout)));
+  for (let i = 0; i < selectedUtxos.length; i++) {
+    updater.addInWitnessUtxo(i + initialNumberOfInputs, selectedUtxos[i].prevout);
+    updater.addInSighashType(i + initialNumberOfInputs, Transaction.SIGHASH_ALL);
+  }
+  updater.addOutputs(
+    recipients.concat(changeOutputs).map((r) => new CreatorOutput(r.asset, r.value, r.address, 0))
+  );
+
```
Comment (tiero) [inline src/application/utils/transaction.ts:287]: you can add inputs and his witnessUtxos & sighash at the same time now with latest LiquidJS

File: src/application/utils/transaction.ts [hunk 2]
```diff
@@ -162,28 +169,53 @@ export async function blindAndSignPset(
...
+    //     return {
+    //       finalScriptSig: undefined,
+    //       finalScriptWitness: witnessStackToScriptWitness([
+    //         ...input.tapScriptSig!.map((s) => s.signature),
+    //         input.tapLeafScript![0].script,
+    //         input.tapLeafScript![0].controlBlock,
+    //       ]),
+    //     };
+    //   });
+    // } else {
```
Comment (tiero) [inline src/application/utils/transaction.ts:185]: what about tapscript spends?

Comment (tiero) [reply src/application/utils/transaction.ts:185]: we should check if tapLeafScript is being set, to determine how to finalize

File: src/application/utils/transaction.ts [hunk 3]
```diff
@@ -274,16 +281,20 @@ export function createTaxiTxFromTopup(
   );
 
   const pset = Pset.fromBase64(taxiReply.topup.partial);
-  const initialNumberOfInputs = pset.inputs.length;
+  const initialNumberOfInputs = pset.globals.inputCount;
```
Comment (tiero) [inline src/application/utils/transaction.ts:284]: whats the difference?

Comment (louisinger) [reply src/application/utils/transaction.ts:285]: no diff but it ensures we have the counter incremented by addInputs methods (if for any reason pset.inputs contains non-input object)

**Comment by tiero** [comment]
Can this be closed?

**Comment by louisinger** [comment]
closed by #436

**Pull request closed without merging**